File: coplur-lead-finder/backend/app/models/lead.py
"""Pydantic models representing a lead / potential Coplur customer."""
from pydantic import BaseModel


class Lead(BaseModel):
    organization_name: str
    website: str | None = None
    industry: str | None = None
    organization_type: str | None = None

    city: str | None = None
    state: str | None = None
    country: str | None = None

    contact_name: str | None = None
    designation: str | None = None
    department: str | None = None

    business_email: str | None = None
    business_phone: str | None = None

    linkedin_url: str | None = None
    organization_linkedin: str | None = None

    source_url: str | None = None

    relevance_score: int = 0
    relevance_reason: str | None = None

File: coplur-lead-finder/frontend/src/types/lead.ts
export interface Lead {
  organization_name: string;
  website?: string | null;
  industry?: string | null;
  organization_type?: string | null;

  city?: string | null;
  state?: string | null;
  country?: string | null;

  contact_name?: string | null;
  designation?: string | null;
  department?: string | null;

  business_email?: string | null;
  business_phone?: string | null;

  linkedin_url?: string | null;
  organization_linkedin?: string | null;

  source_url?: string | null;

  relevance_score: number;
  relevance_reason?: string | null;
}

export interface SearchRequest {
  target_segment: string;
  industry?: string;
  location?: string;
  country?: string;
  keywords: string[];
  roles: string[];
  limit: number;
  min_relevance_score: number;
}

export interface SearchResponse {
  total: number;
  leads: Lead[];
}
